File: src/p0938/python/solution.py
class Solution:
    def rangeSumBST(self, root: TreeNode, l: int, r: int) -> int:
        if not root:
            return 0
        if r < root.val:
            return self.rangeSumBST(root.left, l, r)
        if l > root.val:
            return self.rangeSumBST(root.right, l, r)
        return root.val + self.rangeSumBST(root.left, l, r) + self.rangeSumBST(root.right, l, r)

File: src/p0938/cpp/solution.cpp
class Solution {
public:
    int rangeSumBST(TreeNode *root, int L, int R) {
        if (!root) return 0;
        if (R < root->val) return rangeSumBST(root->left, L, R);
        if (L > root->val) return rangeSumBST(root->right, L, R);
        return root->val + rangeSumBST(root->left, L, R) + rangeSumBST(root->right, L, R);
    }
};
